py/src/ai/h2o/sparkling/ml/algos/H2OSupervisedAlgorithmWithFoldColumn.py

```diff
@@ -0,0 +1,22 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from ai.h2o.sparkling.ml.algos.H2OSupervisedAlgorithm import H2OSupervisedAlgorithm
+
+
+class H2OSupervisedAlgorithmWithFoldColumn(H2OSupervisedAlgorithm):
```

Comment:
would it make more sense to make it a trait? the hierarchy is getting quite complicated

what if you need to add another characteristics/behavior - fold column, stop column... if you have 3+ traits you will not be able to get away with inheritance

Reply:
Yes, it makes sense to use trait on Scala side at first look, but I believe that @satai did it this way because of the logic in [this method](https://github.com/h2oai/sparkling-water/blob/bf7f34571b86bfc45f48a197b87de1947b086c85/ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OSupervisedAlgorithmWithFoldColumn.scala#L35). It expects other fields to be present on the algorithm.

Reply:
thanks for explaining